slyd/app/components/edit-item.js

```diff
@@ -1,41 +1,50 @@
 import Ember from 'ember';
 import NotificationHandler from '../mixins/notification-handler';
 
+export function validateFieldName(name, fields) {
+    // Ensuring that field names start with a letter prevents overwriting
+    // _item, _template and any future "protected" property we might
+    // add to extracted items.
+    if (!/^[a-z]/i.test(name)) {
```

Comment:
Maybe just disallow names beginning with `_`? A user may want to user a number as the first character.